modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v11_2_1/CommercePermissionUpgradeProcess.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.internal.upgrade.v11_2_1;
+
+import com.liferay.account.model.AccountEntry;
+import com.liferay.commerce.constants.CommerceAccountActionKeys;
+import com.liferay.portal.kernel.model.ResourceAction;
+import com.liferay.portal.kernel.model.ResourcePermission;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.ResourceActionLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Stefano Motta
+ */
+public class CommercePermissionUpgradeProcess extends UpgradeProcess {
+
+	public CommercePermissionUpgradeProcess(
+		ResourceActionLocalService resourceActionLocalService,
+		ResourcePermissionLocalService resourcePermissionLocalService) {
+
+		_resourceActionLocalService = resourceActionLocalService;
+		_resourcePermissionLocalService = resourcePermissionLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		ResourceAction resourceAction =
+			_resourceActionLocalService.fetchResourceAction(
+				AccountEntry.class.getName(), ActionKeys.UPDATE);
+
+		if (resourceAction == null) {
+			return;
+		}
+
+		for (ResourcePermission resourcePermission :
+				_resourcePermissionLocalService.getResourcePermissions(
+					AccountEntry.class.getName())) {
+
+			if (_resourcePermissionLocalService.hasActionId(
+					resourcePermission, resourceAction) &&
+				(resourcePermission.getScope() != 4)) {
+
+				_resourcePermissionLocalService.addResourcePermission(
+					resourcePermission.getCompanyId(),
+					AccountEntry.class.getName(), resourcePermission.getScope(),
+					resourcePermission.getPrimKey(),
+					resourcePermission.getRoleId(),
+					CommerceAccountActionKeys.MANAGE_CHANNEL_DEFAULTS);
+				_resourcePermissionLocalService.addResourcePermission(
+					resourcePermission.getCompanyId(),
+					AccountEntry.class.getName(), resourcePermission.getScope(),
+					resourcePermission.getPrimKey(),
+					resourcePermission.getRoleId(),
+					CommerceAccountActionKeys.VIEW_CHANNEL_DEFAULTS);
```

Comment:
what if one has only VIEW? Shouldn't we add the only VIEW permission as per AC-1?

Reply:
If now you don't have the UPDATE permission you cannot view this tab because you are not able to access to the portlets. For this reason I add the VIEW_CHANNEL_DEFAULTS and MANAGE_CHANNEL_DEFAULT permission to all the roles that have UPDATE permission

Reply:
For sure you can "remove" the MANAGE_CHANNEL_DEFAULT permission and you can reach the AC-1 but in the upgrade process I need to maintain the compatibility with the past